### ISSUE TITLE
0.16.5 wont process requests, when using the default port. (no "-p" parameter is given)

### ISSUE THREAD

**Issue by telemaxx** [telemaxx/mapsforgesrv#17]
server starts without errors, but timeout in my frontend app mytourbook.

@nono303: can you see any problems in the startup?
maybe the connector?

![grafik](https://user-images.githubusercontent.com/22706487/148102248-3edf24f7-7e19-4780-8db3-bad3aaddb342.png)

**Comment by telemaxx** [comment]
![grafik](https://user-images.githubusercontent.com/22706487/148104227-03b6d45c-f682-4b78-bcc4-bc29b77c9fee.png)

the server use a random port. in this case 46219

if i set this port in mt, the requests are processed.

this problem is only on linux.

**Comment by JFritzle** [comment]
No problem with my Linux installation!

Using command line:

java -Xmx1G -Xms256M -Xmn256M -jar /mnt/hgfs/D/Austausch/mapsforgesrv-fatjar.neu.jar -if localhost -p 60815 -m "/mnt/hgfs/D/Landkarten/Mapsforge/maps/OpenAndroMaps/Deutschland.map,/mnt/hgfs/D/Landkarten/Mapsforge/maps/OpenAndroMaps/Frankreich Nord.map" -t /home/j/MapsforgeSrv/themes/Elevate/Elevate.xml -s elv-hiking -o elv-h_routes,elv-waymarks,elv-settlements-h,elv-landscapefeat-h,elv-amenities-h,elv-barriers-h,elv-sports-h,elv-accommodation-h,elv-restaurants-h,elv-tourism-h,elv-pubtrans-h,elv-borders-h,elv-background-h -l en -r database -gc 0.72 -cs 0

See attached screenshot and output log.

**Comment by telemaxx** [comment]
now i have found out this:
when i dont give a portnumber the server normaly uses 8080.

this is what i did on my linux.
![grafik](https://user-images.githubusercontent.com/22706487/148106324-7245ec77-ca8e-4193-b1a7-30e6a5b3b5d5.png)
but this is not going to jetty, than jetty uses a random number

but when i give the port number its fine:
![grafik](https://user-images.githubusercontent.com/22706487/148106643-7e4f3e40-f7ee-408a-ba00-39e37fbcb2d9.png)

@JFritzle : can you start the server without the "-p 60815"?
now you should see "8080" several times in the log, but i think you get also random numbers...

**Comment by nono303** [comment]
...This is a bug when port is not set by cmd line or config file
I didn't set the default whatever...
just apply:
```
diff --git "a/mapsforgesrv/src/main/java/com/telemaxx/mapsforgesrv/MapsforgeConfig.java" "b/mapsforgesrv/src/main/java/com/telemaxx/mapsforgesrv/MapsforgeConfig.java"
index 21ccae0..2c7b416 100644
--- "a/mapsforgesrv/src/main/java/com/telemaxx/mapsforgesrv/MapsforgeConfig.java"
+++ "b/mapsforgesrv/src/main/java/com/telemaxx/mapsforgesrv/MapsforgeConfig.java"
@@ -221,6 +221,7 @@ public class MapsforgeConfig {
 			listeningInterface = DEFAULTSERVERINTERFACE;
 		}
 
+		portNumber = DEFAULTSERVERPORT;
 		String portNumberString = retrieveConfigValue("port"); //$NON-NLS-1$
 		if (portNumberString != null) {
 			try {
```
And also, you can simplify:
```
diff --git "a/mapsforgesrv/src/main/java/com/telemaxx/mapsforgesrv/MapsforgeSrv.java" "b/mapsforgesrv/src/main/java/com/telemaxx/mapsforgesrv/MapsforgeSrv.java"
index e1230a7..1d35cc9 100644
--- "a/mapsforgesrv/src/main/java/com/telemaxx/mapsforgesrv/MapsforgeSrv.java"
+++ "b/mapsforgesrv/src/main/java/com/telemaxx/mapsforgesrv/MapsforgeSrv.java"
@@ -84,12 +84,10 @@ public class MapsforgeSrv {
 		}
 		connector.setAcceptQueueSize(mapsforgeConfig.SERVERACCEPTQUEUESIZE);
 		connector.setIdleTimeout(mapsforgeConfig.getIdleTimeout());
-		if (mapsforgeConfig.getListeningInterface().toLowerCase().equals("all")) { //$NON-NLS-1$
-			connector.setPort(mapsforgeConfig.getPortNumber());
-		} else if (mapsforgeConfig.getListeningInterface().toLowerCase().equals("localhost")) { //$NON-NLS-1$
-			connector.setPort(mapsforgeConfig.getPortNumber());
+		connector.setPort(mapsforgeConfig.getPortNumber());
+		if (mapsforgeConfig.getListeningInterface().toLowerCase().equals("localhost")) { //$NON-NLS-1$
 			connector.setHost("127.0.0.1");
-		} else {
+		} else if (!mapsforgeConfig.getListeningInterface().toLowerCase().equals("all")) { //$NON-NLS-1$
 			logger.error("unkown Interface, only \"all\" or \"localhost\" , not " //$NON-NLS-1$
 					+ mapsforgeConfig.getListeningInterface());
 			System.exit(1);
```

**Comment by telemaxx** [comment]
i never used "diff --git"

if you find some time, please send me a PR.

**Comment by nono303** [comment]
> i never used "diff --git"
> 
> if you find some time, please send me a PR.

Ok, I will check all config parser and make the PR

**Comment by telemaxx** [comment]
@nono303 : @JFritzle  is already working on a fix for that bug.
hopefully you have not already invested time ;-)

**Comment by nono303** [comment]
this fix is just to add the line `portNumber = DEFAULTSERVERPORT;` before `String portNumberString = retrieveConfigValue("port");` but i'm actually working on a global refactoring of `MapsforgeConfig` as there is not unified parsing, logging method for all params. 
My PR might come in few hours ;)

**Comment by JFritzle** [comment]
My current MapsforgeConfig version.

**Comment by nono303** [comment]
Just finished https://github.com/telemaxx/mapsforgesrv/pull/18
had check config state with good, with bad and without value in config file and seems OK for all params now ^^
@JFritzle you might just verify `hillShadingMagnitude = (double) parseNumber(DEFAULTHSMAGNITUDE, "hillshading-magnitude", 0., null, "Hillshading magnitude");` in MapsforgeConfig l.468 as I know parse it as a positive double without regexp pattern.

Also reviewed startup logs to be clean on mostly unified:
```
2022-01-05 18:30:09.387 INFO  com.telemaxx.mapsforgesrv.MapsforgeSrv:58@main - MapsforgeSrv - a mapsforge tile server. version: 0.16.5
2022-01-05 18:30:09.393 INFO  com.telemaxx.mapsforgesrv.MapsforgeConfig:458@main - ################## CONFIG INIT ##################
2022-01-05 18:30:09.399 INFO  com.telemaxx.mapsforgesrv.MapsforgeConfig:283@main - Listening TCP port     : defined [8484]
2022-01-05 18:30:09.399 INFO  com.telemaxx.mapsforgesrv.MapsforgeConfig:305@main - Server interface       : defined [all]
2022-01-05 18:30:09.406 INFO  com.telemaxx.mapsforgesrv.MapsforgeConfig:396@main - Map file(s)            : defined [{F:\maps\oam\Alps_oam.osm.map,F:\maps\oam\Belgium_oam.osm.map,F:\maps\oam\Corse_oam.osm.map,F:\maps\oam\France-North_oam.osm.map,F:\maps\oam\France-South_oam.osm.map,F:\maps\oam\Pyrenees_oam.osm.map}]
2022-01-05 18:30:09.407 INFO  com.telemaxx.mapsforgesrv.MapsforgeConfig:335@main - Theme file             : defined [F:\maps\oam\themes\Elevate_5.0.0\Elevate.xml]
2022-01-05 18:30:09.407 INFO  com.telemaxx.mapsforgesrv.MapsforgeConfig:305@main - Theme style            : defined [elv-hiking]
2022-01-05 18:30:09.414 INFO  com.telemaxx.mapsforgesrv.MapsforgeConfig:410@main - Theme overlay(s)       : defined [{elv-h_routes,elv-h_s_routes,elv-waymarks,elv-road_surfaces-h,elv-acc_allowed-h,elv-settlements-h,elv-landscapefeat-h,elv-amenities-h,elv-barriers-h,elv-sports-h,elv-emergency-h,elv-accommodation-h,elv-restaurants-h,elv-shops-h,elv-tourism-h,elv-pubtrans-h,elv-car-h,elv-buildings-h,elv-borders-h,elv-background-h}]
2022-01-05 18:30:09.414 INFO  com.telemaxx.mapsforgesrv.MapsforgeConfig:305@main - Preferred map language : defined [FR]
2022-01-05 18:30:09.415 INFO  com.telemaxx.mapsforgesrv.MapsforgeConfig:283@main - Contrast stretch       : defined [0]
2022-01-05 18:30:09.416 INFO  com.telemaxx.mapsforgesrv.MapsforgeConfig:283@main - Gamma correction       : defined [1.0]
2022-01-05 18:30:09.421 INFO  com.telemaxx.mapsforgesrv.MapsforgeConfig:435@main - HillShading algorithm  : defined [simple(0.0,1.0)]
2022-01-05 18:30:09.422 INFO  com.telemaxx.mapsforgesrv.MapsforgeConfig:283@main - Hillshading magnitude  : defined [1.0]
2022-01-05 18:30:09.433 INFO  com.telemaxx.mapsforgesrv.MapsforgeConfig:335@main - DEM folder             : defined [F:\maps\dem\dtm1]
2022-01-05 18:30:09.433 INFO  com.telemaxx.mapsforgesrv.MapsforgeConfig:305@main - Renderer engine        : defined [direct]
2022-01-05 18:30:09.434 INFO  com.telemaxx.mapsforgesrv.MapsforgeConfig:283@main - Browser cache ttl      : defined [2592000]
2022-01-05 18:30:09.435 INFO  com.telemaxx.mapsforgesrv.MapsforgeConfig:283@main - Server max queue size  : defined [128]
2022-01-05 18:30:09.435 INFO  com.telemaxx.mapsforgesrv.MapsforgeConfig:283@main - Server max thread(s)   : defined [32]
2022-01-05 18:30:09.436 INFO  com.telemaxx.mapsforgesrv.MapsforgeConfig:283@main - Server min thread(s)   : defined [0]
2022-01-05 18:30:09.436 INFO  com.telemaxx.mapsforgesrv.MapsforgeConfig:283@main - Connection idle timeout: defined [120000]
2022-01-05 18:30:09.437 INFO  com.telemaxx.mapsforgesrv.MapsforgeConfig:362@main - Server connector(s)    : defined [{h2c}]
2022-01-05 18:30:09.512 INFO  com.telemaxx.mapsforgesrv.MapsforgeHandler:138@main - ################### MAPS INFO ###################
2022-01-05 18:30:09.521 INFO  com.telemaxx.mapsforgesrv.MapsforgeHandler:145@main - 'F:\maps\oam\Alps_oam.osm.map' supported languages: {de,en,fr,it,sl}
2022-01-05 18:30:09.523 INFO  com.telemaxx.mapsforgesrv.MapsforgeHandler:145@main - 'F:\maps\oam\Belgium_oam.osm.map' supported languages: {de,en,nl,fr}
2022-01-05 18:30:09.524 INFO  com.telemaxx.mapsforgesrv.MapsforgeHandler:145@main - 'F:\maps\oam\Corse_oam.osm.map' supported languages: {de,en,fr}
2022-01-05 18:30:09.525 INFO  com.telemaxx.mapsforgesrv.MapsforgeHandler:145@main - 'F:\maps\oam\France-North_oam.osm.map' supported languages: {de,en,fr}
2022-01-05 18:30:09.526 INFO  com.telemaxx.mapsforgesrv.MapsforgeHandler:145@main - 'F:\maps\oam\France-South_oam.osm.map' supported languages: {de,en,fr}
2022-01-05 18:30:09.527 INFO  com.telemaxx.mapsforgesrv.MapsforgeHandler:145@main - 'F:\maps\oam\Pyrenees_oam.osm.map' supported languages: {de,en,fr,es,gl}
2022-01-05 18:30:09.588 INFO  com.telemaxx.mapsforgesrv.MapsforgeHandler:250@main - ################ THEME FILE INFO ################
2022-01-05 18:30:09.589 INFO  com.telemaxx.mapsforgesrv.MapsforgeHandler:251@main - Default Style: elv-hiking
2022-01-05 18:30:09.590 INFO  com.telemaxx.mapsforgesrv.MapsforgeHandler:253@main - Stylename to use for "-s" option: "elv-hiking" --> RandonnÚes
2022-01-05 18:30:09.590 INFO  com.telemaxx.mapsforgesrv.MapsforgeHandler:253@main - Stylename to use for "-s" option: "elv-city" --> Ville
2022-01-05 18:30:09.591 INFO  com.telemaxx.mapsforgesrv.MapsforgeHandler:253@main - Stylename to use for "-s" option: "elv-cycling" --> Cyclisme
2022-01-05 18:30:09.591 INFO  com.telemaxx.mapsforgesrv.MapsforgeHandler:253@main - Stylename to use for "-s" option: "elv-mtb" --> VÚlo tout terrain
2022-01-05 18:30:09.592 INFO  com.telemaxx.mapsforgesrv.MapsforgeHandler:265@main - The defined style 'elv-hiking' found
2022-01-05 18:30:09.622 INFO  com.telemaxx.mapsforgesrv.MapsforgeHandler:202@Thread-0 - ################# OVERLAY  INFO #################
2022-01-05 18:30:09.624 INFO  com.telemaxx.mapsforgesrv.MapsforgeHandler:215@Thread-0 - 'elv-h_routes' enabled: true" title: '[R] Hiking routes'
2022-01-05 18:30:09.624 INFO  com.telemaxx.mapsforgesrv.MapsforgeHandler:215@Thread-0 - 'elv-h_s_routes' enabled: true" title: '[R] Emphasized routes'
2022-01-05 18:30:09.625 INFO  com.telemaxx.mapsforgesrv.MapsforgeHandler:215@Thread-0 - 'elv-waymarks' enabled: true" title: '[R] Waymarks'
2022-01-05 18:30:09.625 INFO  com.telemaxx.mapsforgesrv.MapsforgeHandler:215@Thread-0 - 'elv-road_surfaces-h' enabled: true" title: '[W] Extended road surfaces'
2022-01-05 18:30:09.626 INFO  com.telemaxx.mapsforgesrv.MapsforgeHandler:215@Thread-0 - 'elv-acc_allowed-h' enabled: true" title: '[W] Marking of permitted footways'
2022-01-05 18:30:09.626 INFO  com.telemaxx.mapsforgesrv.MapsforgeHandler:215@Thread-0 - 'elv-settlements-h' enabled: true" title: '[P] Settlement names'
2022-01-05 18:30:09.627 INFO  com.telemaxx.mapsforgesrv.MapsforgeHandler:215@Thread-0 - 'elv-landscapefeat-h' enabled: true" title: '[P] Landscape features'
2022-01-05 18:30:09.627 INFO  com.telemaxx.mapsforgesrv.MapsforgeHandler:215@Thread-0 - 'elv-amenities-h' enabled: true" title: '[P] Amenities'
2022-01-05 18:30:09.628 INFO  com.telemaxx.mapsforgesrv.MapsforgeHandler:215@Thread-0 - 'elv-barriers-h' enabled: true" title: '[P] Barriers'
2022-01-05 18:30:09.628 INFO  com.telemaxx.mapsforgesrv.MapsforgeHandler:215@Thread-0 - 'elv-sports-h' enabled: true" title: '[P] Sports'
2022-01-05 18:30:09.629 INFO  com.telemaxx.mapsforgesrv.MapsforgeHandler:215@Thread-0 - 'elv-emergency-h' enabled: true" title: '[P] Emergency'
2022-01-05 18:30:09.629 INFO  com.telemaxx.mapsforgesrv.MapsforgeHandler:215@Thread-0 - 'elv-accommodation-h' enabled: true" title: '[P] Accommodation'
2022-01-05 18:30:09.630 INFO  com.telemaxx.mapsforgesrv.MapsforgeHandler:215@Thread-0 - 'elv-restaurants-h' enabled: true" title: '[P] Restaurants + bars'
2022-01-05 18:30:09.630 INFO  com.telemaxx.mapsforgesrv.MapsforgeHandler:215@Thread-0 - 'elv-shops-h' enabled: true" title: '[P] Shops + services'
2022-01-05 18:30:09.631 INFO  com.telemaxx.mapsforgesrv.MapsforgeHandler:215@Thread-0 - 'elv-tourism-h' enabled: true" title: '[P] Tourism + culture'
2022-01-05 18:30:09.631 INFO  com.telemaxx.mapsforgesrv.MapsforgeHandler:215@Thread-0 - 'elv-pubtrans-h' enabled: true" title: '[P] Public transport'
2022-01-05 18:30:09.632 INFO  com.telemaxx.mapsforgesrv.MapsforgeHandler:215@Thread-0 - 'elv-car-h' enabled: true" title: '[P] Car'
2022-01-05 18:30:09.632 INFO  com.telemaxx.mapsforgesrv.MapsforgeHandler:215@Thread-0 - 'elv-buildings-h' enabled: true" title: '[P] Special buildings'
2022-01-05 18:30:09.633 INFO  com.telemaxx.mapsforgesrv.MapsforgeHandler:215@Thread-0 - 'elv-borders-h' enabled: true" title: '[A] Borders + special areas'
2022-01-05 18:30:09.633 INFO  com.telemaxx.mapsforgesrv.MapsforgeHandler:215@Thread-0 - 'elv-background-h' enabled: true" title: '[A] Map background'
2022-01-05 18:30:09.737 INFO  com.telemaxx.mapsforgesrv.MapsforgeSrv:97@main - ################ STARTING SERVER ################
2022-01-05 18:30:09.740 INFO  org.eclipse.jetty.server.Server:383@main - jetty-11.0.7; built: 2021-10-06T19:59:39.004Z; git: 389a3587725d94cc2470ed06aa184c6e57fe9ce8; jvm 17.0.1+12-39
2022-01-05 18:30:09.781 INFO  sun.java2d.marlin:141@Thread-0 - ===============================================================================
2022-01-05 18:30:09.781 INFO  sun.java2d.marlin:141@Thread-0 - Marlin software rasterizer           = ENABLED
2022-01-05 18:30:09.783 INFO  sun.java2d.marlin:141@Thread-0 - Version                              = [marlin-0.9.4.5-Unsafe-OpenJDK]
2022-01-05 18:30:09.783 INFO  sun.java2d.marlin:141@Thread-0 - sun.java2d.renderer                  = sun.java2d.marlin.DMarlinRenderingEngine
2022-01-05 18:30:09.784 INFO  sun.java2d.marlin:141@Thread-0 - sun.java2d.renderer.useThreadLocal   = false
2022-01-05 18:30:09.784 INFO  sun.java2d.marlin:141@Thread-0 - sun.java2d.renderer.useRef           = soft
2022-01-05 18:30:09.785 INFO  sun.java2d.marlin:141@Thread-0 - sun.java2d.renderer.edges            = 4096
2022-01-05 18:30:09.785 INFO  sun.java2d.marlin:141@Thread-0 - sun.java2d.renderer.pixelWidth       = 4096
2022-01-05 18:30:09.786 INFO  sun.java2d.marlin:141@Thread-0 - sun.java2d.renderer.pixelHeight      = 2176
2022-01-05 18:30:09.786 INFO  sun.java2d.marlin:141@Thread-0 - sun.java2d.renderer.profile          = speed
2022-01-05 18:30:09.787 INFO  sun.java2d.marlin:141@Thread-0 - sun.java2d.renderer.subPixel_log2_X  = 8
2022-01-05 18:30:09.787 INFO  sun.java2d.marlin:141@Thread-0 - sun.java2d.renderer.subPixel_log2_Y  = 2
2022-01-05 18:30:09.787 INFO  sun.java2d.marlin:141@Thread-0 - sun.java2d.renderer.tileSize_log2    = 5
2022-01-05 18:30:09.788 INFO  sun.java2d.marlin:141@Thread-0 - sun.java2d.renderer.tileWidth_log2   = 5
2022-01-05 18:30:09.788 INFO  sun.java2d.marlin:141@Thread-0 - sun.java2d.renderer.blockSize_log2   = 5
2022-01-05 18:30:09.789 INFO  sun.java2d.marlin:141@Thread-0 - sun.java2d.renderer.forceRLE         = false
2022-01-05 18:30:09.789 INFO  sun.java2d.marlin:141@Thread-0 - sun.java2d.renderer.forceNoRLE       = false
2022-01-05 18:30:09.789 INFO  sun.java2d.marlin:141@Thread-0 - sun.java2d.renderer.useTileFlags     = true
2022-01-05 18:30:09.790 INFO  sun.java2d.marlin:141@Thread-0 - sun.java2d.renderer.useTileFlags.useHeuristics = true
2022-01-05 18:30:09.791 INFO  sun.java2d.marlin:141@Thread-0 - sun.java2d.renderer.rleMinWidth      = 64
2022-01-05 18:30:09.791 INFO  sun.java2d.marlin:141@Thread-0 - sun.java2d.renderer.useSimplifier    = false
2022-01-05 18:30:09.792 INFO  sun.java2d.marlin:141@Thread-0 - sun.java2d.renderer.usePathSimplifier= false
2022-01-05 18:30:09.797 INFO  sun.java2d.marlin:141@Thread-0 - sun.java2d.renderer.pathSimplifier.pixTol = 0.25
2022-01-05 18:30:09.798 INFO  sun.java2d.marlin:141@Thread-0 - sun.java2d.renderer.clip             = true
2022-01-05 18:30:09.799 INFO  sun.java2d.marlin:141@Thread-0 - sun.java2d.renderer.clip.runtime.enable = false
2022-01-05 18:30:09.799 INFO  sun.java2d.marlin:141@Thread-0 - sun.java2d.renderer.clip.subdivider  = true
2022-01-05 18:30:09.800 INFO  sun.java2d.marlin:141@Thread-0 - sun.java2d.renderer.clip.subdivider.minLength = 100.0
2022-01-05 18:30:09.800 INFO  sun.java2d.marlin:141@Thread-0 - sun.java2d.renderer.doStats          = false
2022-01-05 18:30:09.801 INFO  sun.java2d.marlin:141@Thread-0 - sun.java2d.renderer.doMonitors       = false
2022-01-05 18:30:09.802 INFO  sun.java2d.marlin:141@Thread-0 - sun.java2d.renderer.doChecks         = false
2022-01-05 18:30:09.802 INFO  sun.java2d.marlin:141@Thread-0 - sun.java2d.renderer.useLogger        = true
2022-01-05 18:30:09.803 INFO  sun.java2d.marlin:141@Thread-0 - sun.java2d.renderer.logCreateContext = false
2022-01-05 18:30:09.803 INFO  sun.java2d.marlin:141@Thread-0 - sun.java2d.renderer.logUnsafeMalloc  = false
2022-01-05 18:30:09.806 INFO  sun.java2d.marlin:141@Thread-0 - sun.java2d.renderer.curve_len_err    = 0.01
2022-01-05 18:30:09.807 INFO  sun.java2d.marlin:141@Thread-0 - sun.java2d.renderer.cubic_dec_d2     = 2.5
2022-01-05 18:30:09.807 INFO  sun.java2d.marlin:141@Thread-0 - sun.java2d.renderer.cubic_inc_d1     = 0.5
2022-01-05 18:30:09.808 INFO  sun.java2d.marlin:141@Thread-0 - sun.java2d.renderer.quad_dec_d2      = 1.0
2022-01-05 18:30:09.808 INFO  sun.java2d.marlin:141@Thread-0 - Renderer settings:
2022-01-05 18:30:09.809 INFO  sun.java2d.marlin:141@Thread-0 - MergeSort: DPQS_THRESHOLD: 256
2022-01-05 18:30:09.809 INFO  sun.java2d.marlin:141@Thread-0 - MergeSort: DISABLE_ISORT_THRESHOLD: 1000
2022-01-05 18:30:09.810 INFO  sun.java2d.marlin:141@Thread-0 - SORT         = DPQS_20191112
2022-01-05 18:30:09.811 INFO  sun.java2d.marlin:141@Thread-0 - CUB_DEC_BND  = 640.0
2022-01-05 18:30:09.811 INFO  sun.java2d.marlin:141@Thread-0 - CUB_INC_BND  = 128.0
2022-01-05 18:30:09.812 INFO  sun.java2d.marlin:141@Thread-0 - QUAD_DEC_BND = 256.0
2022-01-05 18:30:09.812 INFO  sun.java2d.marlin:141@Thread-0 - INITIAL_EDGES_CAPACITY               = 98304
2022-01-05 18:30:09.813 INFO  sun.java2d.marlin:141@Thread-0 - INITIAL_CROSSING_COUNT               = 1024
2022-01-05 18:30:09.813 INFO  sun.java2d.marlin:141@Thread-0 - ===============================================================================
2022-01-05 18:30:09.816 INFO  org.eclipse.jetty.server.AbstractConnector:360@main - Started ServerConnector@53fdffa1{HTTP/1.1, (http/1.1, h2c)}{0.0.0.0:8484}
2022-01-05 18:30:09.851 INFO  org.eclipse.jetty.server.Server:438@main - Started Server@242b836{STARTING}[11.0.7,sto=0] @675ms
2022-01-05 18:30:09.855 INFO  com.telemaxx.mapsforgesrv.MapsforgeSrv:104@main - > server connector configured with accept queue size '128', idle timeout '120000'
2022-01-05 18:30:09.857 INFO  com.telemaxx.mapsforgesrv.MapsforgeSrv:105@main - > job executor configured with threads min '0', max '32', idle timeout '120000'
2022-01-05 18:30:09.858 INFO  com.telemaxx.mapsforgesrv.MapsforgeSrv:106@main - > job queue configured with max size '128'
```

**Comment by JFritzle** [comment]
Checking hillShadingMagnitude by
hillShadingMagnitude = (double) parseNumber(DEFAULTHSMAGNITUDE, "hillshading-magnitude", 0., null, "Hillshading magnitude");
seems ok for me, as min value is 0. and max value does not exist.

But I see a problem in line
gammaValue = (double) parseNumber(DEFAULTGAMMA, "gamma-correction", 1., null, "Gamma correction");
First, minimum gamma value is not double value 1. but a double value greater than 0.
Second, if  line gets simply replaced by
gammaValue = (double) parseNumber(DEFAULTGAMMA, "gamma-correction", 0., null, "Gamma correction");
and command line option -gc 0 is given, then an exception will occur later on when evaluating 1./gammaValue. 
Current implementation of function parseNumber is not able to handle error condition value <= minValue but only error condition value < minValue.

**Comment by nono303** [comment]
Hi @JFritzle,
You're right for gamma!
Just pushed availability to exclude min an max value https://github.com/telemaxx/mapsforgesrv/pull/18/commits/43b0223a5b12f9a00eb8d8c6d4add6c3ca15a0ae in https://github.com/telemaxx/mapsforgesrv/pull/18
```
	/*
	 * excludeInRange: true (minValue < value > maxValue) accepted
	 * excludeInRange: false (minValue <= value >= maxValue) accepted
	 */
```
As I see in readme, only gamma need `excludeInRange = true` for `parseNumber`

**Comment by JFritzle** [comment]
Nice. But ...

first: change comment to:
```
	/*
	 * excludeInRange: true (minValue < value < maxValue) accepted
	 * excludeInRange: false (minValue <= value <= maxValue) accepted
	 */
```

second: change line 294 from:
`parseError(msgHeader, "'" + target + "' > '" + maxValue + "' ", defaultValue.toString());`
to
`parseError(msgHeader, "'" + target + operator + maxValue + "' ", defaultValue.toString());`

**Comment by telemaxx** [comment]
@nono303 : can you reduce the output of the embedded log4j.xml / log4.properties a bit?

![grafik](https://user-images.githubusercontent.com/22706487/148407797-9b3a3f92-e3b7-4592-adc3-7ab7e044af8d.png)

for developing we can re enable it with „-Dlog4j.configuration=file:<folder>/log4j.properties“

**Comment by nono303** [comment]
> @nono303 : can you reduce the output of the embedded log4j.xml / log4.properties a bit?
> 
> ![grafik](https://user-images.githubusercontent.com/22706487/148407797-9b3a3f92-e3b7-4592-adc3-7ab7e044af8d.png)
> 
> for developing we can re enable it with „-Dlog4j.configuration=file:/log4j.properties“
https://github.com/telemaxx/mapsforgesrv/pull/18/commits/5c9da4deda4005dd24c02dd023a798132e3ee662
result: `2022-01-06 16:52:50.553 /6/33/26.png?textScale=1.2&userScale=0.8`

```
// log response time thread state and queue size detail for each request
// ex. [ms:1820;idle:0;qs:92]
public boolean LOGREQDET = false;

// log hillshading configuration detail for each request
// ex. SimpleShadingAlgorithm{linearity=0.0, scale=1.0, magnitude=1.0}
public boolean LOGREQDETHS = false;

<!--param name="ConversionPattern" value="%d{yyyy-MM-dd HH:mm:ss.SSS} %-5p %c:%L@%t - %m%n" /-->
<param name="ConversionPattern" value="%d{yyyy-MM-dd HH:mm:ss.SSS} %m%n" />
```

note: 
1. not easy to configure `LOGREQDET` & `LOGREQDETHS` via log4j, I will think about that.
2. const are not final to avoid dead code warning....

**Comment by telemaxx** [comment]
solved in 0.16.6
